packages/repository/src/entity/proto.test.ts

```diff
@@ -3,84 +3,59 @@ import { DeepReadonly } from "./types"
 import { TestEntityData, describe, expect, expectTypeOf, it } from "vitest"
 
 describe("proto", () => {
-  it("Keep type of given input data", ({ syncKeys }) => {
-    const context = createInternalEntity<TestEntityData>(syncKeys)
-
-    const entity = new context({
-      id: "1",
-      foo: "foo",
-      bar: 1,
-    })
-
-    const result = entity.toObject()
-    expect(result).toEqual({
-      foo: "foo",
-      bar: 1,
-      id: "1",
-    })
-    expectTypeOf(result).toEqualTypeOf<
-      DeepReadonly<{ foo: string; bar: number; id: string }>
-    >()
-  })
-
   it("Update method returns new instance with extended data", ({
     syncKeys,
+    fakeData,
```

Comment:
Obczajałem trochę głębiej skąd bierze się ta data i absolutnie ujęła mnie myśl aby wrzucić fakera i takie podstawowe fake'i w `TestContext`. 
Ma to ogromny sens 💯

Dodatkowo nie wiedziałem, że typy np. `TestEntityData` można importować bezprośrednio z vitest.

Reply:
Deklaracje typów każdego modułu możesz rozszerzyć zapisem:
```
declare module "nazwa_modulu" {
// tu ładujesz interfejsy i typy
}
```

Więc są one wtedy dostępne z tego modułu. Warunek jest taki aby plik który zawiera rozszerzenia modułu był w `include` w `tsconfig.ts` - inaczej TS nie będzie kumał o co biega.